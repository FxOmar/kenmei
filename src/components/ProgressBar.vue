<template lang="pug">
  el-progress(:percentage="percentage" :color="colors")
</template>

<script>
  import { Progress } from 'element-ui';

  export default {
    components: {
      'el-progress': Progress,
    },
    props: {
      percentage: {
        type: Number,
        default: 0,
      },
    },
    data() {
      return {
        colors: [
          { color: '#f56c6c', percentage: 20 },
          { color: '#e6a23c', percentage: 40 },
          { color: '#1989fa', percentage: 60 },
          { color: '#6f7ad3', percentage: 80 },
          { color: '#5cb87a', percentage: 100 },
        ],
      };
    },
  };
</script>
