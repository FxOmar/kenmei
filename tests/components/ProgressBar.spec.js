import ProgressBar from '@/components/ProgressBar.vue';

describe('ProgressBar.vue', () => {
  it('renders progress bar', () => {
    const mangaList = shallowMount(ProgressBar);

    expect(mangaList.html()).toMatchSnapshot();
  });
});
