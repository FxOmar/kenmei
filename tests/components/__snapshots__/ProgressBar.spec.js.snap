// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar.vue renders progress bar 1`] = `<el-progress-stub type="line" percentage="0" strokewidth="6" width="126" showtext="true" color="[object Object],[object Object],[object Object],[object Object],[object Object]"></el-progress-stub>`;
